### PR TITLE
Linting js

### PR DESCRIPTION
**Pull request by gagoar**
## Description
We need linting for mono repo working. as requested over at #834 

Some issues with `stylelint`: 

[Here](https://github.com/stylelint/stylelint/issues/5766) it seems to be suggested that some version's combination will do the trick but I tried and it does not work. 

There are several discussions referring to commenting styles that are breaking `stylelint`, `//` breaks it, and `/*` doesn't but I didn't get to test this. 

## Motivation and Context
I feel the need, the need for lint!

## How Has This Been Tested?
I'm running it on my computer, but this PR will hopefully run with the lint workflow. 

## Types of changes

-   [ ] Refactor (changes the way we code something without changing its functionality)
-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

-   [ ] Latest master code has been merged into this branch
-   [ ] No commented out code (if required, place // TODO above with explanation)
-   [ ] No linting issues
-   [ ] Build is successful
-   [ ] Code follows the [API Guidelines](http://britecharts.github.io/britecharts/topics-index.html#toc5__anchor)
-   [ ] Updated the documentation
-   [ ] Added tests to cover changes
-   [ ] All new and existing tests passed

### CODE REVIEW

**Comment by gagoar** [comment]
Have you changed anything? I don't see the workflows running 🤔

**Comment by gagoar** [comment]
> Have you changed anything? I don't see the workflows running 🤔

@Golodhros ^

**Comment by Golodhros** [comment]
> Have you changed anything? I don't see the workflows running 🤔

Nope!
But I see a "Verified" tag on your commits, so I think they are running, right?

File: package.json [hunk 1]
```diff
@@ -121,17 +121,39 @@
...
+    "settings": {
+      "react": {
+        "version": "detect"
+      }
+    },
+    "extends": [
+      "plugin:react/recommended",
+      "plugin:react/jsx-runtime",
+      "plugin:testing-library/react"
+    ],
```
Comment (Golodhros) [inline package.json:133]: Oh, I was surprised to see this here as these are React specific

File: packages/core/src/charts/donut/donut.spec.js [hunk 2]
```diff
@@ -666,8 +666,8 @@ donutDataSets.forEach((datasetName) => {
 
                     donutChart.on('customClick', callback);
                     firstSlice.dispatch('click');
-                    expect(callback.mock.calls.length).toBe(1);
-                    expect(callback.mock.calls[0].length).toBe(3);
+                    expect(callback.mock.calls).toHaveLength(1);
```
Comment (Golodhros) [inline packages/core/src/charts/donut/donut.spec.js:669]: I will probably review this settings, as I do prefer not to use matchers as much as possible, relying on regular JS and a .toEqual or .toBe for everything.

**Review by Golodhros — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (gagoar) [reply package.json:133]: It is needed given that we have JSX in the tests.

File: packages/core/src/charts/donut/donut.spec.js (hunk 2 above)
Comment (gagoar) [reply packages/core/src/charts/donut/donut.spec.js:669]: Some of these changes came from jest recommended rules. I can revert. But it is the recommended way

File: package.json (hunk 1 above)
Comment (Golodhros) [reply package.json:133]: Yeah, but only on the /react package, right?

File: packages/core/src/charts/donut/donut.spec.js (hunk 2 above)
Comment (Golodhros) [reply packages/core/src/charts/donut/donut.spec.js:669]: No prob, I will polish it later on. Not a big deal.

File: package.json (hunk 1 above)
Comment (gagoar) [reply package.json:133]: Yea but liting shares a signle configuation.

**Pull request merged**